src/adaptors/liqwid/index.js

```diff
@@ -1,82 +1,77 @@
+  const endpoint = 'https://v2.api.liqwid.finance/graphql';
+  const registryEndpoint = 'https://public.liqwid.finance/registry.json';
 
-  const query = gql`query ($page: Int) {
-    Page (page: $page) {
-      market {
-        asset {
-          name
-        }
-        state {
-          totalSupply
-          supplyApy
-          utilization
-          borrowApy
-          borrowLqDistributionApy
-          supplyMarketParticipationAPR
-          maxLoanToValue
-        }
-        info {
-          scripts {
-            batchFinal {
-              hash
+  const query = gql`
+    query {
+      liqwid {
+        data {
+          markets {
+            page
+            results {
+              id
+              supplyAPY
+              supply
+              lqSupplyAPY
+              borrowAPY
+              borrow
+              utilization
+              asset {
+                price
+                symbol
+              }
             }
           }
         }
-        price {
-          price
-        }
-        decimals
       }
     }
-    adaStakingAPY
-  }
-  `
-  const data = await request(endpoint, query)
-  const markets = data.Page.market;
-  const adaStakingAPY = data.adaStakingAPY; // ada liquid staking
+  `;
+
+  const data = await request(endpoint, query);
+
+  //NOTE: Action validator will be added to the API in the near future
+  const registryData = await fetch(registryEndpoint)
+    .then((res) => res.json())
+    .then((data) =>
+      data.scriptInfos.filter(
+        (script) =>
+          script.type === 'Validator' && script.tag === 'liqwidMainnet'
+      )
+    );
 
-  const getPool = (tokenName) => {
-    const market = markets.filter((market) => market.asset.name === tokenName)[0];
+  const markets = data.liqwid.data.markets.results;
 
+  const getPool = (market) => {
     return {
-      // FIXME(Kylix, 20th Sep 2023): Construct Shelley address from hex scripthash
-      pool: market.info.scripts.batchFinal.hash.value0, // market batch final hash
-      chain: "Cardano",
-      project: "liqwid",
-      symbol: market.asset.name,
-      tvlUsd: market.state.totalSupply / Math.pow(10, market.decimals) * market.price.price,
-      apyBase: (tokenName == "ADA" ? adaStakingAPY : 0 + Number(market.state.supplyApy)) * 100,
-      apyReward: Number(market.state.supplyMarketParticipationAPR) * 100,
-      rewardTokens: [tokenName, "LQ"],
-      underlyingTokens: [tokenName],
+      pool: registryData.find(
+        (script) => script.name === `Liqwid${market.id}Action`
+      ).scriptHash,
+      chain: 'Cardano',
+      project: 'liqwid',
+      symbol: market.asset.symbol,
+      tvlUsd: market.supply * market.asset.price,
+      apyBase: market.supplyAPY,
+      //NOTE: current API retrieves APY in fraction format but will be changed to percentage in the future
+      apyReward:
+        market.lqSupplyAPY * 100 > 100
+          ? market.lqSupplyAPY
+          : market.lqSupplyAPY * 100,
```

Comment:
This will change for fraction, after our latest discussion internally.
```suggestion
      apyBase: market.supplyAPY * 100,
      apyReward: market.lqSupplyAPY * 100,
```